src/argilla/tasks/datasets/delete.py

```diff
@@ -0,0 +1,34 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TYPE_CHECKING
+
+import typer
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+
+
+def delete_dataset(ctx: typer.Context) -> None:
+    dataset: "RemoteFeedbackDataset" = ctx.obj
+
+    try:
+        dataset.delete()
+    except RuntimeError as e:
```

Comment:
Here we can also handle `PermissionError` which is raised by `allow_for_roles` decorator if the user doesn't have `owner` role